### ISSUE TITLE
Upload from mobile client fails

### ISSUE THREAD

**Issue by mhzawadi** [nextcloud/server#1483]
### Steps to reproduce
1. from Sony album app select image to upload
2. press sharing icon, select nextcloud app
3. select upload folder

### Expected behaviour
The image should be uploaded

### Actual behaviour
nextcloud app error "upload failed"

### Server configuration
**Operating system**: 
nginx - Debian 7.11 (wheezy) on raspberry pi2b+
mysql/PHP-FPM - Debian 8.0 (jessie) on raspberry pi3

**Web server:** 
nginx/1.10.1
**Database:**
MySQL/5.5.50-0+deb8u1
**PHP version:**
5.6.24-0+deb8u1
**Nextcloud version:** (see Nextcloud admin page)
10.0 (stable)
**Updated from an older Nextcloud/ownCloud or fresh install:**
Fresh install
**Where did you install Nextcloud from:**
nextcloud site
**Signing status:**
<details>
<summary>Signing status</summary>

```
No errors have been found.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - activity: 2.3.2
  - admin_audit: 1.0.0
  - bookmarks: 0.9.0
  - calendar: 1.4.0
  - comments: 1.0.0
  - contacts: 1.4.0.0
  - dav: 1.0.0
  - documents: 0.13.1
  - encryption: 1.3.1
  - federatedfilesharing: 1.0.1
  - federation: 1.0.1
  - files: 1.5.2
  - files_accesscontrol: 1.1.2
  - files_automatedtagging: 1.1.1
  - files_external: 1.0.2
  - files_pdfviewer: 0.8.1
  - files_retention: 1.0.0
  - files_sharing: 1.0.0
  - files_texteditor: 2.1
  - files_trashbin: 1.0.0
  - files_versions: 1.3.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 15.0.0
  - mail: 0.5.3
  - notifications: 0.3.0
  - password_policy: 1.0.0
  - provisioning_api: 1.0.0
  - serverinfo: 1.1.1
  - survey_client: 0.1.5
  - systemtags: 1.0.2
  - tasks: 0.9.3
  - templateeditor: 0.1
  - theming: 1.0.1
  - twofactor_totp: true
  - updatenotification: 1.0.1
  - user_external: 0.4
  - workflowengine: 1.0.1
Disabled:
  - external
  - user_ldap
  - user_saml
```
</details>

**The content of config/config.php:**
<details>
<summary>Config report</summary>

```
{
    "system": {
        "instanceid": "oc0xc6xq5cff",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "owncloud.horwood.biz",
            "212.159.117.222"
        ],
        "datadirectory": "\/var\/data\/horwood_owncloud",
        "overwrite.cli.url": "https:\/\/owncloud.horwood.biz",
        "dbtype": "mysql",
        "version": "9.1.0.16",
        "dbname": "horwood_owncloud",
        "dbhost": "192.168.49.205",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "logfile": "\/var\/data\/owncloud\/owncloud.log",
        "logdateformat": "Y-m-d H:i:s",
        "installed": true,
        "mail_smtpmode": "smtp",
        "mail_from_address": "owncloud",
        "mail_domain": "horwood.biz",
        "mail_smtphost": "vps.horwood.biz",
        "mail_smtpport": "25",
        "theme": "horwood_mini",
        "filelocking.enabled": "true",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "192.168.49.206",
            "port": 6379,
            "timeout": 0
        },
        "asset-pipeline.enabled": false,
        "filesystem_check_changes": "0",
        "trashbin_retention_obligation": "auto",
        "preview_max_scale_factor": 1,
        "enabledPreviewProviders": {
            "0": "OC\\Preview\\PNG",
            "1": "OC\\Preview\\JPEG",
            "2": "OC\\Preview\\GIF",
            "11": "OC\\Preview\\Illustrator",
            "12": "OC\\Preview\\Postscript",
            "13": "OC\\Preview\\Photoshop",
            "14": "OC\\Preview\\TIFF"
        },
        "maintenance": false,
        "loglevel": 2,
        "htaccess.RewriteBase": "\/"
    }
}
```
</details>

**Are you using external storage, if yes which one:** local

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** NO

#### LDAP configuration (delete this part if not used)
<details>
<summary>LDAP config</summary>

```
[Symfony\Component\Console\Exception\CommandNotFoundException]
  There are no commands defined in the "ldap" namespace.
```
</details>

### Client configuration
**Browser:** Nextcloud app 1.3.1 from play store

**Operating system:** Android 6.0.1

#### Web server error log
<details>
<summary>Web server error log</summary>
```
192.168.49.7 - anna [21/Sep/2016:20:50:33 +0100] "GET /ocs/v1.php/apps/files_sharing/api/v1/shares?path=%2F&reshares=true&subfiles=true HTTP/1.1" 200 138 "-" "owncloud.horwood.biz" "Mozilla/5.0 (Android) ownCloud-android/1.3.1" "-" "192.168.49.205:9000"
192.168.49.7 - anna [21/Sep/2016:20:50:33 +0100] "HEAD /remote.php/webdav/Seth/ HTTP/1.1" 200 0 "-" "owncloud.horwood.biz" "Mozilla/5.0 (Android) ownCloud-android/1.3.1" "-" "192.168.49.206:9000"
192.168.49.7 - anna [21/Sep/2016:20:50:33 +0100] "GET /index.php/avatar/anna/384 HTTP/1.1" 200 31 "-" "owncloud.horwood.biz" "Mozilla/5.0 (Android) ownCloud-android/1.3.1" "-" "192.168.49.206:9000"
192.168.49.7 - anna [21/Sep/2016:20:50:36 +0100] "HEAD /remote.php/webdav/Seth/DSC_2425.JPG.jpg HTTP/1.1" 503 0 "-" "owncloud.horwood.biz" "Mozilla/5.0 (Android) ownCloud-android/1.3.1" "-" "192.168.49.205:9000"
192.168.49.7 - anna [21/Sep/2016:20:50:39 +0100] "PUT /remote.php/webdav/Seth/DSC_2425.JPG.jpg-chunking-14384609-3-0 HTTP/1.1" 500 252 "-" "owncloud.horwood.biz" "Mozilla/5.0 (Android) ownCloud-android/1.3.1" "-" "192.168.49.206:9000"
192.168.49.6 - - [21/Sep/2016:20:50:46 +0100] "GET /ocs/v2.php/apps/notifications/api/v1/notifications?format=json HTTP/1.1" 200 85 "-" "owncloud.horwood.biz" "Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.116 Safari/537.36 OPR/39.0.2256.71" "-" "192.168.49.205:9000"
192.168.49.6 - matt [21/Sep/2016:20:50:55 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 420 "-" "owncloud.horwood.biz" "Mozilla/5.0 (Windows) mirall/2.2.3 (build 4) (Nextcloud)" "-" "192.168.49.206:9000"
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>
```
{"reqId":"mvr+No3BZXwUz31KXniP","remoteAddr":"192.168.49.7","app":"webdav","message":"Exception: {\"Message\":\"HTTP\\\/1.1 503 Storage not available\",\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\ServiceUnavailable\",\"Code\":0,\"Trace\":\"#0 \\\/mnt\\\/data1\\\/var\\\/data\\\/websites\\\/3rd_party\\\/horwood_nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(79): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\ObjectTree->getNodeForPath('Seth\\\/DSC_2425.J...', 0)\\n#1 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpGet(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#2 \\\/mnt\\\/data1\\\/var\\\/data\\\/websites\\\/3rd_party\\\/horwood_nextcloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#3 \\\/mnt\\\/data1\\\/var\\\/data\\\/websites\\\/3rd_party\\\/horwood_nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(459): Sabre\\\\Event\\\\EventEmitter->emit('method:GET', Array)\\n#4 \\\/mnt\\\/data1\\\/var\\\/data\\\/websites\\\/3rd_party\\\/horwood_nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(251): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response), false)\\n#5 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpHead(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#6 \\\/mnt\\\/data1\\\/var\\\/data\\\/websites\\\/3rd_party\\\/horwood_nextcloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#7 \\\/mnt\\\/data1\\\/var\\\/data\\\/websites\\\/3rd_party\\\/horwood_nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(459): Sabre\\\\Event\\\\EventEmitter->emit('method:HEAD', Array)\\n#8 \\\/mnt\\\/data1\\\/var\\\/data\\\/websites\\\/3rd_party\\\/horwood_nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(248): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#9 \\\/mnt\\\/data1\\\/var\\\/data\\\/websites\\\/3rd_party\\\/horwood_nextcloud\\\/apps\\\/dav\\\/appinfo\\\/v1\\\/webdav.php(60): Sabre\\\\DAV\\\\Server->exec()\\n#10 \\\/mnt\\\/data1\\\/var\\\/data\\\/websites\\\/3rd_party\\\/horwood_nextcloud\\\/remote.php(165): require_once('\\\/mnt\\\/data1\\\/var\\\/...')\\n#11 {main}\",\"File\":\"\\\/mnt\\\/data1\\\/var\\\/data\\\/websites\\\/3rd_party\\\/horwood_nextcloud\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/ObjectTree.php\",\"Line\":162,\"User\":\"anna\"}","level":4,"time":"2016-09-21 19:50:36","method":"HEAD","url":"\/remote.php\/webdav\/Seth\/DSC_2425.JPG.jpg","user":"anna"}
{"reqId":"OfRCOfAJ45MSDFPuxF\/F","remoteAddr":"192.168.49.7","app":"webdav","message":"Exception: {\"Message\":\"File with id \\\"3201\\\" has not been found.\",\"Exception\":\"OCP\\\\Files\\\\StorageNotAvailableException\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/data\\\/websites\\\/3rd_party\\\/horwood_nextcloud\\\/lib\\\/private\\\/Files\\\/Storage\\\/Wrapper\\\/Jail.php(214): OC\\\\Files\\\\Storage\\\\FailedStorage->file_exists('\\\/DSC_2425.JPG.j...')\\n#1 \\\/var\\\/data\\\/websites\\\/3rd_party\\\/horwood_nextcloud\\\/lib\\\/private\\\/Files\\\/Storage\\\/Wrapper\\\/Wrapper.php(220): OC\\\\Files\\\\Storage\\\\Wrapper\\\\Jail->file_exists('DSC_2425.JPG.jp...')\\n#2 \\\/var\\\/data\\\/websites\\\/3rd_party\\\/horwood_nextcloud\\\/lib\\\/private\\\/Files\\\/View.php(1120): OC\\\\Files\\\\Storage\\\\Wrapper\\\\Wrapper->file_exists('DSC_2425.JPG.jp...')\\n#3 \\\/var\\\/data\\\/websites\\\/3rd_party\\\/horwood_nextcloud\\\/lib\\\/private\\\/Files\\\/View.php(525): OC\\\\Files\\\\View->basicOperation('file_exists', '\\\/Seth\\\/DSC_2425....')\\n#4 \\\/var\\\/data\\\/websites\\\/3rd_party\\\/horwood_nextcloud\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/Directory.php(259): OC\\\\Files\\\\View->file_exists('\\\/Seth\\\/DSC_2425....')\\n#5 \\\/var\\\/data\\\/websites\\\/3rd_party\\\/horwood_nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Tree.php(105): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\Directory->childExists('DSC_2425.JPG.jp...')\\n#6 \\\/var\\\/data\\\/websites\\\/3rd_party\\\/horwood_nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(504): Sabre\\\\DAV\\\\Tree->nodeExists('Seth\\\/DSC_2425.J...')\\n#7 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpPut(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#8 \\\/var\\\/data\\\/websites\\\/3rd_party\\\/horwood_nextcloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#9 \\\/var\\\/data\\\/websites\\\/3rd_party\\\/horwood_nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(459): Sabre\\\\Event\\\\EventEmitter->emit('method:PUT', Array)\\n#10 \\\/var\\\/data\\\/websites\\\/3rd_party\\\/horwood_nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(248): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#11 \\\/var\\\/data\\\/websites\\\/3rd_party\\\/horwood_nextcloud\\\/apps\\\/dav\\\/appinfo\\\/v1\\\/webdav.php(60): Sabre\\\\DAV\\\\Server->exec()\\n#12 \\\/var\\\/data\\\/websites\\\/3rd_party\\\/horwood_nextcloud\\\/remote.php(165): require_once('\\\/var\\\/data\\\/websi...')\\n#13 {main}\",\"File\":\"\\\/var\\\/data\\\/websites\\\/3rd_party\\\/horwood_nextcloud\\\/lib\\\/private\\\/Files\\\/Storage\\\/FailedStorage.php\",\"Line\":111,\"User\":\"anna\"}","level":4,"time":"2016-09-21 19:50:39","method":"PUT","url":"\/remote.php\/webdav\/Seth\/DSC_2425.JPG.jpg-chunking-14384609-3-0","user":"anna"}
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>
```
Mobile app (no logs sorry)
```
</details>

**Comment by mhzawadi** [comment]
I have downloaded the f-droid app and that also fails to upload, I have pinned down the issue to a shared folder. If that helps.

**Comment by nickvergessen** [comment]
@AndyScherzinger any idea maybe?

**Comment by AndyScherzinger** [comment]
@mhzawadi which f-droid version of the Nextcloud client did you install? Beta or "normal/stable", the beta should have a log (accessible in the apps settings menu). Can you see anything in there regarding your upload issue?

**Comment by mhzawadi** [comment]
Hi @AndyScherzinger, I had the stable version installed. But have removed and installed the beta to get the logs.

have checked the file limit and I get 'You are using 6.3 GB of Unlimited' on the account.

[currentLog.txt](https://github.com/nextcloud/server/files/504444/currentLog.txt)

**Comment by mhzawadi** [comment]
Hi @AndyScherzinger I have found the issue!

So I have a share with my wife call Seth (our sons pics), some how it got resetup and called Seth (2). This seems to cause the upload to fail.

So a have truncated the share table and setup new shares, this seems to have fixed it.

**Comment by AndyScherzinger** [comment]
@mhzawadi glad to here you could resolve the matter. Besides I think this is related to #1588 which has been resolved and is fixed in Version 11 and most likely 10.0.2, right @MorrisJobke ?

**Comment by MorrisJobke** [comment]
> @mhzawadi glad to here you could resolve the matter. Besides I think this is related to #1588 which has been resolved and is fixed in Version 11 and most likely 10.0.2, right @MorrisJobke ?

Yes, the backport is opened :)
